plugins/modules/cloud/misc/proxmox_snap.py

```diff
@@ -0,0 +1,289 @@
+                                 proxmox.nodes(vm[0]['node']).tasks(taskid).log.get()[:1])
+
+        time.sleep(1)
+    return False
+
+
+def snapshot_remove(module, proxmox, vm, vmid, timeout, snapname, force):
+    taskid = getattr(proxmox.nodes(vm[0]['node']), VZ_TYPE)(vmid).snapshot.delete(snapname, force=int(force))
+    while timeout:
+        if (proxmox.nodes(vm[0]['node']).tasks(taskid).status.get()['status'] == 'stopped' and
+                proxmox.nodes(vm[0]['node']).tasks(taskid).status.get()['exitstatus'] == 'OK'):
+            return True
+        timeout -= 1
+        if timeout == 0:
+            module.fail_json(msg='Reached timeout while waiting for removing VM snapshot. Last line in task before timeout: %s' %
+                                 proxmox.nodes(vm[0]['node']).tasks(taskid).log.get()[:1])
+
+        time.sleep(1)
+    return False
+
+
+def setup_api(api_host, api_user, api_password, validate_certs):
+    api = ProxmoxAPI(api_host, user=api_user, password=api_password, verify_ssl=validate_certs)
+    return api
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            api_host=dict(required=True),
+            api_user=dict(required=True),
+            api_password=dict(no_log=True),
+            vmid=dict(required=False),
+            validate_certs=dict(type='bool', default='no'),
+            hostname=dict(),
+            timeout=dict(type='int', default=30),
+            state=dict(default='present', choices=['present', 'absent']),
+            description=dict(type='str'),
+            snapname=dict(type='str', default='ansible_snap'),
+            force=dict(type='bool', default='no'),
+            vmstate=dict(type='bool', default='no'),
+        )
+    )
+
+    if not HAS_PROXMOXER:
+        module.fail_json(msg='proxmoxer required for this module')
+
+    state = module.params['state']
+    api_user = module.params['api_user']
+    api_host = module.params['api_host']
+    api_password = module.params['api_password']
+    vmid = module.params['vmid']
+    validate_certs = module.params['validate_certs']
+    hostname = module.params['hostname']
+    description = module.params['description']
+    snapname = module.params['snapname']
+    timeout = module.params['timeout']
+    force = module.params['force']
+    vmstate = module.params['vmstate']
+
+    # If password not set get it from PROXMOX_PASSWORD env
+    if not api_password:
+        try:
+            api_password = os.environ['PROXMOX_PASSWORD']
+        except KeyError as e:
+            module.fail_json(msg='You should set api_password param or use PROXMOX_PASSWORD environment variable')
+
+    try:
+        proxmox = setup_api(api_host, api_user, api_password, validate_certs)
+
+    except Exception as e:
+        module.fail_json(msg='authorization on proxmox cluster failed with exception: %s' % e)
```

Comment:
```suggestion
        module.fail_json(msg='authorization on proxmox cluster failed with exception: %s' % to_native(e))
```
this prevents python-version-related errors

Reply:
gotcha! fixed! :-)